smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpProtocolClientGenerator.kt

```diff
@@ -94,7 +94,7 @@ abstract class HttpProtocolClientGenerator(
      * or else you need to override [renderInit] and construct it manually
      */
     protected open val defaultHttpClientEngineSymbol: Symbol = buildSymbol {
-        name = "KtorEngine"
+        name = "DefaultHttpClientEngine"
         namespace(KotlinDependency.HTTP_KTOR_ENGINE)
     }
```

Comment:
**Question**: Is an overridable definition still required here? Or now that we have an `expect` definition can we just move this to **RuntimeTypes.kt** like all our other static symbols?

Reply:
Yes for now we still need this. `expect` / `actual` doesn't apply to dependent code. They are resolved at compile time for the given sourceSet. IIRC there is something on the kotlin roadmap to maybe allow that sort of thing where you can say I expect you to provide this but it doesn't exist right now.


Eventually I want this to be removed and rely on #495